go-controller/pkg/clustermanager/subnetallocator/host_subnet_allocator.go

```diff
@@ -12,6 +13,7 @@ import (
 )
 
 type HostSubnetAllocator struct {
+	sync.Mutex
```

Comment:
are we using this anywhere for locking?

Reply:
Yes.  This was suggested by @martinkennelly 

Please see this as an example - https://github.com/ovn-org/ovn-kubernetes/pull/3127/files#diff-2f72aac69e673565092afb59f7bf845e20a162b9f9e8651ed03458a3b2812d43R71